### PR TITLE
Create events page

### LINKED ISSUE #123: Create events page
**The event specific page is a different task ( #270 ) **
```[tasklist]
### Tasks
- [x] Create event card
- [x] Display all events card
- [x] Sort based on date
- [x] Don't show events that have ended
```

### PR DESCRIPTION
**Pull request by maarcelh**
Create event page

* Closes #123

### CODE REVIEW

File: public/images/manifest.json
Comment (fkoulen) [inline public/images/manifest.json:1]: Move this back please.

Comment (maarcelh) [reply public/images/manifest.json:1]: woops

File: src/components/events/event-card.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const startDateTime = startDate.getHours() + ":" + startDate.getMinutes()
+
+    if (endTimeStamp != null) {
+      endDate = new Date(endTimeStamp)
+      const endDateTime = endDate.getHours() + ":" + endDate.getMinutes()
+      return `${startDateTime} - ${endDateTime}`
+    }
+
+    return startDateTime
+  }
```
Comment (fkoulen) [inline src/components/events/event-card.tsx:31]: You can do this easily with format of date-fns
This renders Wednesday, June 14 for example:

https://github.com/ASVGay/the-rhapsodies/blob/4d0ee3b13a48edab26fc15a5fda3a4494fab1ef3/src/components/events/event-info-card.tsx#L28 

You can also check that file to see how I did it with the time. See the [docs of date-fns](https://date-fns.org/v2.30.0/docs/format) for more examples

File: src/components/events/event-card.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const date = new Date(timestamp)
+    return `${DayMapper[date.getDay()]}, ${MonthMapper[date.getMonth()]} ${date.getDate()}`
+  }
+
+  const getStartAndEndTime = (startTimeStamp: string, endTimeStamp: string | null) => {
+    let endDate
+    const startDate = new Date(startTimeStamp)
+    const startDateTime = startDate.getHours() + ":" + startDate.getMinutes()
+
+    if (endTimeStamp != null) {
```
Comment (fkoulen) [inline src/components/events/event-card.tsx:24]: Why did you make this nullable?

File: src/pages/events/index.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      .then((res) => {
+        if (res.error) {
+          setErrorText("Failed to load events, try refreshing the page.")
+          return
+        }
+
+        if (res.data?.length > 0) {
+          setEvents(res.data as Event[])
+        } else {
+          setErrorText("No Events have been added yet.")
```
Comment (fkoulen) [inline src/pages/events/index.tsx:28]: events*

File: src/pages/events/index.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  const [events, setEvents] = useState<Event[]>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(true)
+  const [errorText, setErrorText] = useState("")
+  const fetchEvents = () => {
+    setShowSpinner(true)
+    const events = getEventsWithAttendees(supabaseClient)
+    events
+      .then((res) => {
+        if (res.error) {
+          setErrorText("Failed to load events, try refreshing the page.")
```
Comment (fkoulen) [inline src/pages/events/index.tsx:21]: Users on mobile can't refresh the page so I would just keep it at failed to load events

File: src/pages/events/index.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+const Index = () => {
+  const supabaseClient = useSupabaseClient<Database>()
+  const [events, setEvents] = useState<Event[]>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(true)
+  const [errorText, setErrorText] = useState("")
+  const fetchEvents = () => {
+    setShowSpinner(true)
+    const events = getEventsWithAttendees(supabaseClient)
+    events
```
Comment (fkoulen) [inline src/pages/events/index.tsx:18]: Could just inline this

File: src/components/events/event-card.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      return `${startDateTime} - ${endDateTime}`
+    }
+
+    return startDateTime
+  }
+
+  return (
+    <div className="h-fit w-[22rem] cursor-pointer overflow-hidden rounded-md bg-white drop-shadow-lg">
+      <div className="w-full pb-1">
+        <Image className={"object-cover object-bottom h-[120px]"} src={getEventImage(event.event_type)} alt={"Event Image"} width={358} height={121}/>
```
Comment (fkoulen) [inline src/components/events/event-card.tsx:36]: Removing the object-bottom will center the image. Looks a bit better imo :)

File: src/components/events/attendance-card.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import {CheckCircleIcon, QuestionMarkCircleIcon, XCircleIcon} from "@heroicons/react/24/solid";
```
Comment (fkoulen) [inline src/components/events/attendance-card.tsx:2]:
```suggestion
import {CheckCircleIcon, QuestionMarkCircleIcon, XCircleIcon} from "@heroicons/react/20/solid";
```

File: src/types/database.ts [hunk 8]
```diff
@@ -28,6 +34,40 @@ export interface Database {
...
+          remark?: string | null
+          user_id: string
+        }
+        Update: {
+          attending?: Database["public"]["Enums"]["attending"]
+          event_id?: string
+          remark?: string | null
+          user_id?: string
+        }
+        Relationships: [
```
Comment (rensdoodeman) [inline src/types/database.ts:56]: Why do you need these? Other table relationships haven't been defined like this

File: src/components/events/event-card.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+          <div className={"flex flex-row gap-2"}>
+            <CalendarIcon height={20} width={20} color={"#EEC73F"} />
+            <span className={"text-base leading-5 font-light text-zinc-400"}>
+               {format(new Date(event.start_time), "cccc, LLLL d")}
+            </span>
+          </div>
+          <div className={"flex flex-row gap-2"}>
+            <ClockIcon height={20} width={20} color={"#EEC73F"} />
+            <span className={"text-base leading-5 font-light text-zinc-400"}>
+              {format(new Date(event.start_time), "HH:mm")} -{" "}
```
Comment (rensdoodeman) [inline src/components/events/event-card.tsx:34]: Can't you save these as dates?

Comment (maarcelh) [reply src/components/events/event-card.tsx:34]: wym

File: src/types/database.ts (hunk 8 above)
Comment (maarcelh) [reply src/types/database.ts:56]: this is autogenerated

**Review by fkoulen — APPROVED** [review]

**Pull request merged**